### DISCUSSION TITLE
Using CLI/Bulk Import with half of my library already uploaded via iOS app?

### DISCUSSION THREAD

**Discussion by Normanras** [immich-app/immich#5688] [Q&A] [answered]
I have about 40k photos and while I was aware of the CLI bulk upload, I wanted to put the iOS app through its paces with the foreground/background processes and uploading photos. I am about 80% through the upload, which so far has been a positive experience. However, since I am self-hosting Immich, my ISP modem is not happy and now the state table fills up (and thus stops all new connections to my network) in about 30 minutes or less. This is with the app open, the phone unlocked, and the foreground process running. 

Since I got most of the way there and only have ~6k photos left, I'm satisfied with the app experience and am considering just using the bulk upload tool. 

Will Immich recognize the already uploaded photos and avoid duplicates? Or is the bulk upload tool best used for fresh installs?

I know [Immich-go](https://github.com/simulot/immich-go) is also recommended by the immich devs, but since it's community-maintained I'm curious (but not expecting) if anyone knows if Immich-go will also recognize duplicates in the upload process. I'm aware the go says remove duplicates but it seems to be only based on the file name and I'm unsure if iOS > Google Photo preserves the same name. 

Thanks in advance!

**Comment by bo0tzz** [comment, marked as answer]
Immich hashes files as they get uploaded and checks those against the database. If a hash already exists, it will not allow the file to be uploaded, thus preventing duplicates.

**Reply by Normanras** [reply]
Thanks so much!
